djautotask/api.py

```diff
@@ -251,7 +251,8 @@ def __init__(
         integration_code=None,
         rest_api_version=None,
         server_url=None,
-        impersonation_id=None,
+        impersonation_resource=None,
+        impersonation_target=None,
```

Comment:
Instead of using Model classes in `init_limited_impersonation_targets`, you should use API client classes. Then you could remove this `impersonation_target` argument. The client would be able to use itself, rather than relying on being passed the Model class that it doesn't actually do anything with. The API classes are 1-to-1 with models, so this should be a very easy change.

Reply:
I looked into that, but found some not-working cases.

We call API's w/ two different patterns for now. 

1. From Synchronizer such as `self.sctr_synchronizer(impersonation_resource=r)`, which has api client_class and it is good to go.

2. But, we have other use cases by 'save' or 'update_at' from model, which doesn't have api client_class.
e.g. 
```
card.save(impersonation_resource=r, update_at=True,
                      changed_fields=form.changed_data)
```
```
        return card.update_at(impersonation_resource=r,
                              changed_fields=updated_fields)
```

So, Using model could be consistent method to approach.

Reply:
Never mind... It seems possible.. Will do that.

Reply:
I appreciate your review again. Good to see the improvement.